staking/programs/staking/src/lib.rs

```diff
@@ -564,17 +562,88 @@ pub mod staking {
      * the config account. If accepted, `amount` tokens are transferred to a new stake account
      * owned by the `recipient` and the split request is reset (by setting `amount` to 0).
      * The recipient of a transfer can't vote during the epoch of the transfer.
+     *
+     * The `pda_authority` must explicitly approve both the amount of tokens and recipient, and
+     * these parameters must match the request (in the `split_request` account).
      */
-    pub fn accept_split(ctx: Context<AcceptSplit>) -> Result<()> {
-        // TODO : Split vesting schedule between both accounts
+    pub fn accept_split(ctx: Context<AcceptSplit>, amount: u64, recipient: Pubkey) -> Result<()> {
+        let config = &ctx.accounts.config;
+        config.check_frozen()?;
 
-        // TODO : Transfer stake positions to the new account if need
+        let current_epoch = get_current_epoch(config)?;
+
+        let split_request = &ctx.accounts.source_stake_account_split_request;
+        require!(
+            split_request.amount == amount && split_request.recipient == recipient,
+            ErrorCode::InvalidApproval
+        );
+
+        // Initialize new accounts
+        ctx.accounts.new_stake_account_metadata.initialize(
+            *ctx.bumps.get("new_stake_account_metadata").unwrap(),
+            *ctx.bumps.get("new_stake_account_custody").unwrap(),
+            *ctx.bumps.get("new_custody_authority").unwrap(),
+            *ctx.bumps.get("new_voter_record").unwrap(),
+            &split_request.recipient,
+            Some(current_epoch),
```

Comment:
What do you want to do about `transfer_epoch`

Reply:
Set it to None maybe?

Reply:
ooh good catch thanks